components/brave_wallet_ui/utils/state-migration-utils.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { Reducer } from '@reduxjs/toolkit'
+import {
+  PersistedState,
+  PersistConfig,
+  persistReducer
+} from 'redux-persist'
+
+import storage from 'redux-persist/lib/storage'
+import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
+
+/**
+ * Increment this version number when:
+ * - removing queries form the api slice
+ * - removing state from the redux store
+ */
+export const PERSISTED_STATE_VERSION = 1
+
+/**
+ * Uses the enhanced config object to persist a reducer
+ * with type safe blacklist & whitelist
+ * 
+ * Clears the persisted state when the version changes
+ * to prevent storing old/unused data
+ * 
+ * @param reducer The reducer to persist
+ * @param config options for how to persist the state
+ * @returns persisted reducer
+ */
+export const persistVersionedReducer = <
+  R extends Reducer,
+  C extends Omit<PersistConfig<ReturnType<R>>, 'storage'> & {
+    blacklist?: Array<keyof ReturnType<R>>
+    whitelist?: Array<keyof ReturnType<R>>
+  }
+>(
+  reducer: R,
+  config: C
+): Reducer<PersistedState & ReturnType<R>> => {
+  const revertToInitialStateOnVersionChange = async <
+    T extends PersistedState & ReturnType<R>
+  >(
+    state: T
+  ): Promise<T> => {
+    // remove persisted API state when persisted state versions change
```

Comment:
can persisted API state also be removed when wallet is disabled / deleted / reset?

Reply:
I have logged an issue to add the ability to clear this data when the wallet is reset: https://github.com/brave/brave-browser/issues/29138

Reply:
Wallet local storage will now be cleared on wallet reset as of: https://github.com/brave/brave-core/pull/17809